File: Projekat/fronend/models/proizvod.js

export class Proizvod {

    constructor(id, naziv, cena, dostupan, putanja, idTipaProizvoda){
        if(id)
            this.id=id;
        else
            this.id=-1;

        if(naziv)
            this.naziv=naziv;
        else
            this.naziv="";
        
        if(cena)
            this.cena=cena;
        else
            this.cena=-1;

        if(dostupan)
            this.dostupan=dostupan;
        else
            this.dostupan=false;

        if(putanja)
            this.putanja=putanja;
        else
            this.putanja="";
            
        if(idTipaProizvoda)
            this.idTipaProizvoda=idTipaProizvoda;
        else
            this.idTipaProizvoda
    }
}

File: Projekat/backend/Models/Proizvod.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore.Infrastructure;
using System.Text.Json.Serialization;

namespace Models{

    public class Proizvod{

        [Key]
        public int Id {get; set;}

        [Required]
        [MaxLength(50)]
        public string Naziv {get; set; }

        [JsonIgnore]
        public TipProizvoda TipProizvoda {get; set; }

        [JsonIgnore]
        public Pijaca Pijaca {get; set; }

        public int Cena {get; set; }

        public bool Dostupan {get; set; }

        public string Putanja {get; set;}
        
    }
}